ECIES switch to Electrum-ECIES

monkeylord opened the pull request:
Consider Electrum is most widely used client, we should follow Electrum-ECIES, rather than old Bitcore-ECIES.
However, Bitcore-ECIES is still preserved for compatiblity. It's at ECIES.bitcoreECIES.

ryanxcharles commented:
in the future it would be good if you simply updated the same PR, so that the conversation about why certain things happen can be maintained. there's no reason to close and re-open- just commit your changes to the existing PR.

previous PR: https://github.com/moneybutton/bsv/pull/21

ryanxcharles commented on lib/ecies/electrum-ecies.js:
```diff
@@ -0,0 +1,150 @@
+// Encrypts the message (String or Buffer).
+ECIES.prototype.encrypt = function (message, ivbuf) {
+  if (!Buffer.isBuffer(message)) message = Buffer.from(message)
+  var ciphertext = AESCBC.encrypt(message, this.kE, this.iv)
```
here is where you simply want to use the .encryptCipherkey method instead of .encrypt ... that way, the original API can stay the same, and we can still distinguish the two different methods.

monkeylord replied:
ivbuf here is unused. will delete.

ryanxcharles commented on lib/ecies/electrum-ecies.js:
```diff
@@ -0,0 +1,150 @@
+    throw new errors.DecryptionError('Invalid checksum')
+  }
+
+  return AESCBC.decrypt(ciphertext, this.kE, this.iv)
```
here this should use .decryptCipherkey

monkeylord replied:
IV is used here.
So it may be decryptCipherkeyIV and IV, or decrypt, I think.

monkeylord commented:
Actually, I've checked that in old bitcore "aescbc.js", encrypt/decrypt is never used.
So I merged my "aes128cbc.js" and "aescbc.js" and removed unused functions.

The encrypt/decrypt function is brand new.
1. removed unused encrypt/decrypt
2. add new encrypt/decrypt from aes128cbc.js

The reason why I kept encryptCipherkey/decryptCipherkey is because they are not standard.
The message they encrypt/decrypt have IV appended.
So I have to leave them there.

monkeylord commented on test/ecies/ecies.js:
```diff
@@ -3,8 +3,8 @@
 var ECIES = require('../../lib/ecies')
```
changed.

ryanxcharles commented:
awesome! will review shortly
